torch/csrc/autograd/engine.cpp

```diff
@@ -723,27 +771,48 @@ auto Engine::execute(const edge_list& roots,
 }
 
 void Engine::enqueue_blocked_task_on_cpu(NodeTask task) {
-  std::call_once(start_threads_flag_, &Engine::start_threads, this);
-  ready_queue(at::kCPU).push(
+  std::call_once(start_device_threads_flag_, &Engine::start_device_threads, this);
+  std::shared_ptr<GraphTask> graph_task = task.base_.lock();
+  TORCH_INTERNAL_ASSERT(graph_task, "GraphTask is no longer valid!");
```

Comment:
It would be good to have a brief argument why `graph_task` must be live at this point